### PR TITLE
Fix bug

### PR DESCRIPTION
**Pull request by tonywu315**
When `null_replication_metrics` is enabled and any columns is all null, this errors:
```Python
import dataprofiler as dp
import pandas as pd
import re

profiler_options = dp.ProfilerOptions()

NO_FLAG = 0
profiler_options.set({
    '*.null_values': {
        "": NO_FLAG,
        "nan": re.IGNORECASE,
        "none": re.IGNORECASE,
        "null": re.IGNORECASE,
        "  *": NO_FLAG,
        "--*": NO_FLAG,
        "__*": NO_FLAG,
        "9"*7: NO_FLAG,
    }, 
    "*.null_replication_metrics.is_enabled": True,
    "data_labeler.is_enabled": False,
    "multiprocess.is_enabled": False,
})
profiler = dp.Profiler(pd.DataFrame([[9999999,9], [9999999,9]]), options=profiler_options)
```

### CODE REVIEW

File: dataprofiler/profilers/profile_builder.py [hunk 1]
```diff
@@ -2244,7 +2244,10 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
         ]._profiles[get_data_type(profile)]
 
         total_row_sum = np.asarray(
-            [get_data_type_profiler(profile).sum for profile in self._profile]
+            [
+                get_data_type_profiler(profile).sum if get_data_type(profile) else 0
```
Comment (JGSweets) [inline dataprofiler/profilers/profile_builder.py:2248]: I think np.nan might be more appropriate.

File: dataprofiler/profilers/profile_builder.py [hunk 2]
```diff
@@ -2290,7 +2293,11 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
             sum_not_null = np.delete(total_row_sum, col_id) - sum_null
 
             mean_null = sum_null / null_count
-            mean_not_null = sum_not_null / true_count
+
+            if true_count == 0:
+                mean_not_null = sum_not_null
+            else:
+                mean_not_null = sum_not_null / true_count
```
Comment (JGSweets) [inline dataprofiler/profilers/profile_builder.py:2300]: is this necessary?

**Comment by JGSweets** [comment]
```python
profiler = dp.Profiler(pd.DataFrame([[9999999,9], [9999999,9]]), options=profiler_options)
```
Is the erroring I meant to put.

File: dataprofiler/profilers/profile_builder.py [hunk 3]
```diff
@@ -2244,7 +2244,10 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
         ]._profiles[get_data_type(profile)]
 
         total_row_sum = np.asarray(
-            [get_data_type_profiler(profile).sum for profile in self._profile]
```
[the remark this answers is not in the document]
Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py:2247]: The test can be as simple as running the example code you have in the description and ensuring the rep matrix is what we expect

File: dataprofiler/profilers/profile_builder.py (hunk 2 above)
Comment (tonywu315) [reply dataprofiler/profilers/profile_builder.py:2300]: When a column is all null, `true_count` is 0, which creates a divide by 0 error.

**Comment by taylorfturner** [comment]
@tonywu315 needs and `update branch`. Will review one updated

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 4]
```diff
@@ -2048,6 +2048,47 @@ def test_null_replication_metrics_calculation(self):
...
+        column = report["data_stats"][0]["null_replication_metrics"]
+
+        self.assertTrue(len(column["class_prior"]) == 2)
+        self.assertTrue(len(column["class_sum"]) == 2)
+        self.assertTrue(len(column["class_mean"]) == 2)
+
+        np.testing.assert_array_almost_equal([0, 1], column["class_prior"])
+
+        np.testing.assert_array_almost_equal([0], column["class_sum"][0])
+        np.testing.assert_array_almost_equal([18], column["class_sum"][1])
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:2087]: could this be simplified to the below?
```
np.testing.assert_array_almost_equal([[0],[18]], column["class_sum"])
```

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 5]
```diff
@@ -2048,6 +2048,47 @@ def test_null_replication_metrics_calculation(self):
...
+
+        np.testing.assert_array_almost_equal([0], column["class_mean"][0])
+        np.testing.assert_array_almost_equal([9], column["class_mean"][1])
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:2090]: same as `class_sum`: could we simplify this to one line?

**Review by taylorfturner — COMMENTED** [review]
just two comments around simplifying the test suite that is added in this PR

**Review by taylorfturner — APPROVED** [review]
LGTM

File: dataprofiler/profilers/profile_builder.py [hunk 6]
```diff
@@ -2290,7 +2295,11 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
             sum_not_null = np.delete(total_row_sum, col_id) - sum_null
 
             mean_null = sum_null / null_count
-            mean_not_null = sum_not_null / true_count
+
+            if true_count == 0:
+                mean_not_null = sum_not_null
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2300]: technically this isn't a `mean` value ... its a `summation` value, though?

Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py:2300]: i'd suggest doing a:
```python
mean_not_null = np.nan
 if true_count:
    mean_not_null = sum_not_null / true_count
```

Comment (taylorfturner) [reply dataprofiler/profilers/profile_builder.py:2300]: +1 @tony

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: dataprofiler/profilers/profile_builder.py [hunk 7]
```diff
@@ -2373,7 +2391,11 @@ def _merge_null_replication_metrics(self, other: StructuredProfiler) -> Dict:
...
+            else:
+                mean_not_null = sum_not_null / true_count
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply dataprofiler/profilers/profile_builder.py:2398]:
```
mean_not_null = np.nan
 if true_count:
    mean_not_null = sum_not_null / true_count
```

**Review by taylorfturner — APPROVED** [review]
LGTM

**Review by taylorfturner — DISMISSED** [review]

**Comment by taylorfturner** [comment]
@tonywu315 `update branch` FYI

File: dataprofiler/profilers/profile_builder.py [hunk 8]
```diff
@@ -2290,7 +2295,12 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
             sum_not_null = np.delete(total_row_sum, col_id) - sum_null
 
             mean_null = sum_null / null_count
-            mean_not_null = sum_not_null / true_count
+
+            if true_count == 0:
+                sum_not_null = np.full(len(data.columns) - 1, np.nan)
+                mean_not_null = np.full(len(data.columns) - 1, np.nan)
+            else:
+                mean_not_null = sum_not_null / true_count
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2303]: Why not? 
```
sum_not_null = np.full(len(data.columns) - 1, np.nan)
mean_not_null = np.full(len(data.columns) - 1, np.nan)

 if true_count:
    mean_not_null = sum_not_null / true_count
```

File: dataprofiler/profilers/profile_builder.py [hunk 9]
```diff
@@ -2373,7 +2393,12 @@ def _merge_null_replication_metrics(self, other: StructuredProfiler) -> Dict:
             sum_not_null = np.delete(total_row_sum, col_id) - sum_null
 
             mean_null = sum_null / null_count
-            mean_not_null = sum_not_null / true_count
+
+            if true_count == 0:
+                sum_not_null = np.full(len(self._profile) - 1, np.nan)
+                mean_not_null = np.full(len(self._profile) - 1, np.nan)
+            else:
+                mean_not_null = sum_not_null / true_count
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2401]: thoughts on? 
```
sum_not_null = np.full(len(data.columns) - 1, np.nan)
mean_not_null = np.full(len(data.columns) - 1, np.nan)

 if true_count:
    mean_not_null = sum_not_null / true_count
```

**Review by taylorfturner — COMMENTED** [review]
code was changed back to a prior paradigm... can we keep that same paradigm but just update first assignment?

File: dataprofiler/profilers/profile_builder.py (hunk 8 above)
Comment (tonywu315) [reply dataprofiler/profilers/profile_builder.py:2303]: `sum_not_null` needs to be in an if because its value is updated before and not after

Comment (taylorfturner) [reply dataprofiler/profilers/profile_builder.py:2303]: so could we do 
```
mean_not_null = np.full(len(data.columns) - 1, np.nan)

 if true_count:
    sum_not_null = np.full(len(data.columns) - 1, np.nan)
    mean_not_null = sum_not_null / true_count
```

Comment (taylorfturner) [reply dataprofiler/profilers/profile_builder.py:2303]:
```
mean_not_null = np.full(len(data.columns) - 1, np.nan)
if not true_count:
    sum_not_null = np.full(len(data.columns) - 1, np.nan)
else:
    mean_not_null = sum_not_null / true_count
```

**Review by taylorfturner — APPROVED** [review]
LGMT

File: dataprofiler/profilers/profile_builder.py [hunk 10]
```diff
@@ -2373,7 +2393,12 @@ def _merge_null_replication_metrics(self, other: StructuredProfiler) -> Dict:
             sum_not_null = np.delete(total_row_sum, col_id) - sum_null
 
             mean_null = sum_null / null_count
-            mean_not_null = sum_not_null / true_count
+
+            mean_not_null = np.full(len(self._profile) - 1, np.nan)
+            if not true_count:
+                sum_not_null = np.full(len(self._profile) - 1, np.nan)
+            else:
+                mean_not_null = sum_not_null / true_count
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2401]: this how this code was previously
```
            sum_not_null = np.delete(total_row_sum, col_id) - sum_null

            mean_null = sum_null / null_count
            mean_not_null = sum_not_null / true_count
```
there is a concern that `np.full()` is getting run after `np.delete()`. Can we validate that is not materially changing anything downstream. This comes across as a larger change to the source code.

File: dataprofiler/profilers/profile_builder.py [hunk 11]
```diff
@@ -2290,7 +2295,12 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
...
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2303]: same as below

File: dataprofiler/profilers/profile_builder.py [hunk 12]
```diff
@@ -2290,7 +2295,12 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
             sum_not_null = np.delete(total_row_sum, col_id) - sum_null
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2295]: so this is doing this type of operation... 
```
In [1]: import numpy as np

In [2]: arr = np.array([[1,2,3,4], [5,6,7,8], [9,10,11,12]])

In [3]: arr
Out[3]: 
array([[ 1,  2,  3,  4],
       [ 5,  6,  7,  8],
       [ 9, 10, 11, 12]])

In [4]: np.delete(arr, 0,0)
Out[4]: 
array([[ 5,  6,  7,  8],
       [ 9, 10, 11, 12]])
```

File: dataprofiler/profilers/profile_builder.py [hunk 13]
```diff
@@ -2290,7 +2295,12 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
...
 
             mean_null = sum_null / null_count
-            mean_not_null = sum_not_null / true_count
+
+            mean_not_null = np.full(len(self._profile) - 1, np.nan)
+            if not true_count:
+                sum_not_null = np.full(len(self._profile) - 1, np.nan)
```
Comment (taylorfturner) [inline dataprofiler/profilers/profile_builder.py:2301]: if `true_count` is zero (`if not true_count`)...which sets `sum_not_null` to a single array ... not an `array` of `arrays` like `sum_not_null` on L2295 returns...

File: dataprofiler/profilers/profile_builder.py (hunk 12 above)
Comment (taylorfturner) [reply dataprofiler/profilers/profile_builder.py:2295]: although I maybe setting up this wrong...

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 14]
```diff
@@ -2048,6 +2048,39 @@ def test_null_replication_metrics_calculation(self):
         np.testing.assert_array_almost_equal([17 / 8, 48 / 8], column["class_mean"][0])
         np.testing.assert_array_almost_equal([12 / 2, 6 / 2], column["class_mean"][1])
 
+        # Test with all null in a column
+        data_3 = pd.DataFrame([[9999999, 9], [9999999, 9]])
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:2052]: @tonywu315 I'm actually running a debugger through here and `true_count` never seems to actually trigger ```sum_not_null = np.full(len(self._profile) - 1, np.nan)``` CC @JGSweets

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/tests/profilers/test_profile_builder.py (hunk 14 above)
Comment (taylorfturner) [reply dataprofiler/tests/profilers/test_profile_builder.py:2052]: mea culpa

**Review by JGSweets — APPROVED** [review]

File: dataprofiler/tests/profilers/test_profile_builder.py (hunk 14 above)
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_profile_builder.py:2052]: in the future, might be worthwhile to create a dataset with more edge cases 
`data_3 = pd.DataFrame([[9999999, 9999999, 123], [9999999, 123, 123]])`
